test/gtest/ucp/test_ucp_am.cc

```diff
@@ -1267,3 +1267,41 @@ UCS_TEST_P(test_ucp_am_nbx_rndv_memtype, rndv)
 }
 
 UCP_INSTANTIATE_TEST_CASE_GPU_AWARE(test_ucp_am_nbx_rndv_memtype);
+
+
+class test_ucp_am_nbx_rndv_memtype_disable_zcopy :
+        public test_ucp_am_nbx_rndv_memtype {
+protected:
+    void disable_rndv_zcopy_config(entity &e, uint64_t caps)
+    {
+        ucp_ep_h ep             = e.ep();
+        ucp_ep_config_t *config = ucp_ep_config(ep);
+
+        if (caps == UCT_IFACE_FLAG_PUT_ZCOPY) {
+            ucp_ep_config_rndv_zcopy_commit(0, &config->rndv.put_zcopy);
+        } else {
+            ASSERT_TRUE(caps == UCT_IFACE_FLAG_GET_ZCOPY);
+            ucp_ep_config_rndv_zcopy_commit(0, &config->rndv.get_zcopy);
+        }
+    }
+};
+
+UCS_TEST_P(test_ucp_am_nbx_rndv_memtype_disable_zcopy, rndv_disable_put_zcopy)
+{
+    disable_rndv_zcopy_config(sender(), UCT_IFACE_FLAG_PUT_ZCOPY);
+    disable_rndv_zcopy_config(receiver(), UCT_IFACE_FLAG_PUT_ZCOPY);
+
+    ucs_memory_type_t mt = static_cast<ucs_memory_type_t>(get_variant_value(0));
+    test_am_send_recv(64 * UCS_KBYTE, 8, 0, mt);
+}
+
+UCS_TEST_P(test_ucp_am_nbx_rndv_memtype_disable_zcopy, rndv_disable_get_zcopy)
+{
+    disable_rndv_zcopy_config(sender(), UCT_IFACE_FLAG_GET_ZCOPY);
+    disable_rndv_zcopy_config(receiver(), UCT_IFACE_FLAG_GET_ZCOPY);
+
+    ucs_memory_type_t mt = static_cast<ucs_memory_type_t>(get_variant_value(0));
+    test_am_send_recv(64 * UCS_KBYTE, 8, 0, mt);
+}
```

Comment:
can define a common function for these tests with caps parameter

Reply:
done